src/runtime/write_debug_image.cpp

```diff
@@ -167,7 +207,75 @@ WEAK extern "C" int halide_debug_to_file(void *user_context, const char *filenam
 
     uint32_t final_padding_bytes = 0;
 
-    if (ends_with(filename, ".tiff") || ends_with(filename, ".tif")) {
+    if (ends_with(filename, ".npy")) {
+        npy_dtype_info_t di = {0, 0, 0};
+        for (const auto &d : npy_dtypes) {
+            if (d.htype == buf->type) {
+                di = d.dtype;
+                break;
+            }
+        }
+        if (di.byte_order == 0) {
+            halide_error(user_context, "Unsupported element type for .npy in debug_to_file()\n");
+            return halide_error_code_debug_to_file_failed;
+        }
+
+        constexpr int kMaxDictStringSize = 1024;
```

Comment:
We don't typically use this naming convention for constants. max_dict_string_size would be more in keeping with the rest of the compiler.